server/parser/src/main/java/org/eclipse/lsp/cobol/AntlrRangeUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+/**
+ * The AntlrRangeUtils class provides a utility method to construct Range objects.
+ */
+public class AntlrRangeUtils {
+    /**
+     * Construct the range from ANTLR context
+     *
+     * @param ctx the ANTLR context
+     * @return the range
+     */
+    public static Range constructRange(ParseTree ctx) {
+        if (ctx instanceof ParserRuleContext) {
+            return constructRange((ParserRuleContext) ctx);
+        }
+        if (ctx instanceof TerminalNode) {
+            return constructRange(((TerminalNode) ctx).getSymbol());
+        }
+        throw new AssertionError("Unexpected token: " + ctx.getText());
+    }
+
+    /**Ï
+     * Construct the range from ANTLR context
+     *
+     * @param ctx the ANTLR context
+     * @return the range
+     */
+    public static Range constructRange(ParserRuleContext ctx) {
+      Token start = ctx.start;
+      Token stop = ctx.stop;
+      Position startPosition = new Position(start.getLine() - 1, start.getCharPositionInLine());
+      Position stopPosition = ctx.stop == null || start.getLine() > stop.getLine()
```

Comment:
```suggestion
      Position stopPosition = ctx.stop == null || start.getStartIndex() >= stop.getStopIndex()
```
?

Reply:
good one. Are you sure about ">="? I think "=" case is a valid one character token/rule.

Reply:
I think you are correct, it should only test for `>`. The antlr start/stop are inclusive.